### ISSUE TITLE
Save Class Probabilities during Inference

### ISSUE THREAD

**Issue by ramppdev** [autrainer/autrainer#39]
During inference prediction, the `results.csv` outputs the filename, optional offset (sliding window), prediction, and output (list of logits) for each sample.
This should be extended to include the output probabilities as well.
- classification: include a separate column for each class in the dataset
- regression: include a separate column for each target (re-scaled): this requires the target transform to have names for the target similar to classification. We currently do not yet support multi-target regression but this may be good to add then as well.

**Comment by ATriantafyllopoulos** [comment]
I anyways wanted to add multi-target regression, so that's a nice addition anyway (would need to adapt the encoder a bit, I presume).

To achieve this, I think we need to extend [_predict](https://github.com/autrainer/autrainer/blob/d1ca7efbb2a42961ebf4313ef8546e8b1856678a/autrainer/serving/serving.py#L324) to also return something like this: `{key: float(output[0, index].numpy()) for index, key in enumerate(self.target_transform.labels)}`

However, this depends on #40

**Comment by ramppdev** [comment]
During inference we have the following pipeline (M = Model, T = target transform): `M.outputs` -> `T.predict_batch` -> `T.decode` (map) -> (decoded).
During output tracking we currently call  `M.outputs` -> `T.predict_batch` -> `T.decode` (map) -> (save dataframe).

We could extend both pipelines to include e.g.  `M.outputs` -> `T.probabilities_batch` -> `T.predict_batch` -> ... with `T.probabilities_batch` applying the following operations:
- (single-target) regression: identity OP; still scaled (no update of `T.predict_batch` required)
- multi-class classification: Softmax (no update of `T.predict_batch` required; we only argmax the outputs)
- multi-label classification: Sigmoid (remove Sigmoid from `T.predict_batch`)

Then in all cases we can save the outputs as well as the result of `T.probabilities_batch` (+ destructure the probabilities to separate columns in the inference `results.csv`).

> To achieve this, I think we need to extend [_predict](https://github.com/autrainer/autrainer/blob/d1ca7efbb2a42961ebf4313ef8546e8b1856678a/autrainer/serving/serving.py#L324) to also return something like this: `{key: float(output[0, index].numpy()) for index, key in enumerate(self.target_transform.labels)}`

We then can adapt the target transforms to destructure outputs and probabilities as well, simplifying inference here,.
For  (multi-target) regression this would require saving the name of the target column (probably makes most sense).

**Comment by ramppdev** [comment]
> Yes, that's the plan I had in mind as well! Do you have time to prepare a PR?

Yes i will try to implement it soon.
